xapian-letor/include/xapian-letor/feature.h

```diff
@@ -55,19 +104,39 @@ class XAPIAN_VISIBILITY_DEFAULT Feature {
     /// Virtual destructor because we have virtual methods.
     virtual ~Feature();
 
-    /// Specify the database to use for feature building. This will be used by the Internal class.
+    /// Specify the database to use for feature building. This will be used by FeatureList::Internal class.
```

Comment:
"by the FeatureList::Internal class" would read better.

This (and similar) doc comments should really be split.  The first line should be a short summary (the first sentence here is ideal), then there should be a blank (except for the comment `*`) line, and then more detail (the "This will be used ..." part can go there).

I'd say "This is used ..." though, as least if this describes the current situation.  Using the future tense suggests to the reader that it might not currently work that way, but is intended to after some planned code changes.